docker-images/kafka-based/kafka/scripts/zookeeper_run.sh

```diff
@@ -37,6 +37,22 @@ export CERTS_STORE_PASSWORD
 
 mkdir -p /tmp/zookeeper
 
+# If it should run a check on DNS name resolving before starting ZooKeeper
+# This is useful in environment (i.e. minikube) where DNS registration is slow
+# It is also related to https://issues.apache.org/jira/browse/ZOOKEEPER-4708
+if [ -z "$ZOOKEEPER_DNS_CHECKS_DISABLED" ] || [ "$ZOOKEEPER_DNS_CHECKS_DISABLED" = "false" ]; then
+  echo "Resolving ${BASE_HOSTNAME}-$((ZOOKEEPER_ID-1)).${BASE_FQDN} ..."
+  ipaddress=$(nslookup ${BASE_HOSTNAME}-$((ZOOKEEPER_ID-1)).${BASE_FQDN} | grep "Address" | awk '{print $2}' | sed -n 2p)
+
+  while [ -z $ipaddress ]; do
+    sleep 1
+    echo "Resolving ${BASE_HOSTNAME}-$((ZOOKEEPER_ID-1)).${BASE_FQDN} ..."
+    ipaddress=$(nslookup ${BASE_HOSTNAME}-$((ZOOKEEPER_ID-1)).${BASE_FQDN} | grep "Address" | awk '{print $2}' | sed -n 2p)
+  done
+fi
+
+echo "IP address $ipaddress"
```

Comment:
I think this will make it a bit more understandable for users => that you are not resolving it multiple times but you are tryin to resolve it until it does actually resolve.

```suggestion
if [ -z "$ZOOKEEPER_DNS_CHECKS_DISABLED" ] || [ "$ZOOKEEPER_DNS_CHECKS_DISABLED" = "false" ]; then
  echo "Trying to resolve ${BASE_HOSTNAME}-$((ZOOKEEPER_ID-1)).${BASE_FQDN} ..."
  ipaddress=$(nslookup ${BASE_HOSTNAME}-$((ZOOKEEPER_ID-1)).${BASE_FQDN} | grep "Address" | awk '{print $2}' | sed -n 2p)

  while [ -z $ipaddress ]; do
    sleep 1
    echo "Trying to resolve ${BASE_HOSTNAME}-$((ZOOKEEPER_ID-1)).${BASE_FQDN} ..."
    ipaddress=$(nslookup ${BASE_HOSTNAME}-$((ZOOKEEPER_ID-1)).${BASE_FQDN} | grep "Address" | awk '{print $2}' | sed -n 2p)
  done
fi

echo "Resolved IP address $ipaddress"
```